src/main/java/bio/terra/service/filedata/azure/util/BlobContainerClientFactory.java

```diff
@@ -22,6 +24,11 @@
 
   public static final Duration DELEGATED_KEY_DURATION = Duration.ofHours(24);
   private final HttpClient httpClient = HttpClient.createDefault();
+  private static final int MAX_RETRIES = 3;
+  private static final int RETRY_TIMEOUT_SECONDS = 3600;
+  private final RequestRetryOptions retryOptions =
+      new RequestRetryOptions(
+          RetryPolicyType.EXPONENTIAL, MAX_RETRIES, RETRY_TIMEOUT_SECONDS, null, null, null);
```

Comment:
Nice! So, this is built-in retry functionality?

Reply:
Yeah! Jesus commented on it here with some more info https://github.com/DataBiosphere/jade-data-repo/pull/1027#discussion_r684401754